compiler/grainc/dune

```diff
@@ -3,4 +3,4 @@
  (public_name grainc)
  (package grain)
  (modules grainc)
- (libraries grain dune-build-info))
+ (libraries grain binaryen.native dune-build-info))
```

Comment:
Is this necessary?

Reply:
It's not strictly required because we have binaryen.ml use the native lib by default, but I like being explicit because we ship the js lib with it now too.

Reply:
I meant like, is this necessary to specify for `grainc`?

Reply:
(Explained, and I get it now 👍)